db/migrate/20231215165741_add_guests_table.rb

```diff
@@ -0,0 +1,11 @@
+class AddGuestsTable < ActiveRecord::Migration[7.0]
+  def change
+    create_table :guests do |t|
+      t.string :name
+      t.integer :nights
+      t.references :stay, null: false, foreign_key: true
```

Comment:
A guest can stay in multiple rooms, rather than just one (it's half of the many-to-many) so we *don't* want to include the stay reference here. A guest can exist on its own without any `stay` or `room` columns